compiler-core/src/type_/environment.rs

```diff
@@ -482,11 +482,17 @@ impl<'a> Environment<'a> {
             .entity_usages
             .pop()
             .expect("Expected a bottom level of entity usages.");
-        self.handle_unused(unused);
+        self.handle_unused(unused.clone());
 
         for (name, location) in self.unused_modules.clone().into_iter() {
-            self.warnings
-                .emit(Warning::UnusedImportedModule { name, location });
+            // Check if the module is really unused
+            if !unused
+                .iter()
+                .any(|(_, (_, loc, used))| *used && location.contains(loc.start))
```

Comment:
What is this doing? To me it seems something is off if we have to do this additional checking, and I don't understand exactly what this is from the code and the comment.

Reply:
This is checking if one of the unqualified name is used. I agree it seems something is off. I guess the issue is that `as name` usage needs to be handled independently, e.g. by using a new `UnusedImportedModuleName` warning. To emit the right location, could we add this new struct:

```rust
pub struct ImportName {
  pub name: SmolStr,
  pub location: SrcSpan,
}
```

... for the `as_name` field of the Import:

```diff
  pub struct Import<PackageName> {
    pub documentation: Option<SmolStr>,
    pub location: SrcSpan,
    pub module: SmolStr,
-   pub as_name: Option<SmolStr>,
+   pub as_name: Option<ImportName>,
    pub unqualified: Vec<UnqualifiedImport>,
    pub package: PackageName,
}
```